### ISSUE TITLE
not working for async rest method with CompletableFuture

### ISSUE THREAD

**Issue by kirmandi** [sercasti/spring-httpserver-timings#10]
using this library, even without having a method of async type annotated with @Traceable, results always in empty content.

see https://spring.io/guides/gs/async-method/ for async method with CompletableFuture example.

any endpoint just returns no content and also has content-length 0, seems like the interceptor is not working correctly here.

**Comment by michalrrcc** [comment]
I am facing the same issue. Any update on the fix @sercasti ?

**Comment by sercasti** [comment]
Yes, as soon as I come back from my vacations I'll work on this, feel free
to post a PR
